src/linear_algebra/affine_space.lean

```diff
@@ -15,19 +15,71 @@ open_locale classical
 # Affine spaces
 
 This file defines affine spaces (over modules) and subspaces, affine
-maps, affine combinations of points, and the affine span of a set of
-points.
+maps, affine combinations of points, affinely independent families of
+points, and the affine span of a set of points.
+
+## Main definitions
+
+* `affine_space k V P` is an abbreviation for `add_torsor V P` in the
+  case of `module k V`.  Definitions and results not depending on the
+  `module` structure appear in `algebra.add_torsor` instead of here;
+  that includes the instance of an `add_group` as an `add_torsor` over
+  itself, which thus gives a `module` as an `affine_space` over
+  itself.  Definitions of affine spaces vary as to whether a space
+  with no points is permitted; here, we require a nonempty type of
+  points (via the definition of torsors requiring a nonempty type).
+  Affine spaces are defined over any module, with stronger type class
+  requirements on `k` being used for individual lemmas where needed.
+* `affine_subspace k V P` is the type of affine subspaces.  Unlike
+  affine spaces, affine subspaces are allowed to be empty, and lemmas
+  that do not apply to empty affine subspaces have `nonempty`
+  hypotheses.  There is a `complete_lattice` structure on affine
+  subspaces.
+* `affine_subspace.direction` gives the `submodule` spanned by the
+  pairwise differences of points in an `affine_subspace`.  There are
+  various lemmas relating to the set of points in the `direction`, and
+  relating the lattice structure on affine subspaces to that on their
+  directions.
+* `affine_span` gives the affine subspace spanned by a set of points,
+  with `vector_span` giving its direction.  `affine_span` is defined
+  in terms of `span_points`, which gives an explicit description of
+  the points contained in the affine span; `span_points` itself should
+  generally only be used when that description is required, with
+  `affine_span` being the main definition for other purposes.  Two
+  other descriptions of the affine span are proved equivalent: it is
+  the `Inf` of affine subspaces containing the points, and (if
+  `[nontrivial k]`) it contains exactly those points that are affine
+  combinations of points in the given set.
+* `weighted_vsub_of_point`, `weighted_vsub` and `affine_combination`
+  are various kinds of weighted combinations of points.
+  `weighted_vsub_of_point` is a general weighted combination of
+  subtractions with an explicit base point, yielding a vector.
+  `weighted_vsub` uses a default choice of base point and is intended
+  to be used when the sum of weights is 0, in which case the result is
+  independent of the choice of base point.  `affine_combination` adds
```

Comment:
```suggestion
  `weighted_vsub` uses an arbitrary choice of base point and is intended
  to be used when the sum of weights is 0, in which case the result is
  independent of the choice of base point.  `affine_combination` adds
```
The word "default" seems to imply that the point comes from `inhabited.default` instead of `classical.choice`.